lib/imagereader/imagereader.h

```diff
@@ -27,10 +28,15 @@ class ImageReader
     static std::unique_ptr<ImageReader> createFdiImageReader(const ImageReaderProto& config);
 
 public:
-	virtual Image readImage() = 0;
+	const Image& readImage();
+	void preloadImage();
+
+protected:
+	virtual std::unique_ptr<Image> readImageImpl() = 0;
```

Comment:
Is the std::unique_ptr just there to reduce copies or does something else require it?

Reply:
It's just to remove copies.